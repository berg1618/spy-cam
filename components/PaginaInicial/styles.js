import { StyleSheet } from "react-native";

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#726E6E"
  },
  titleApp: {
    fontSize: 40,
    color: '#F2F0F4',
    fontWeight: '800',
    // marginLeft: 20
  },
  Modal: {
    position: "relative",
    left: 60,
    top: 4,
    width: 33,
    height: 29
  },
  header: {
    marginTop: 15,
    height: 87,
    flexDirection: "row",
    alignItems: "center",
    borderBottomColor: '#fff',
    borderBottomWidth: 2,
    justifyContent: "center"
  },
  main: {
    alignItems: "center",
    marginTop: 54,
  },

  imgLogo: {
    width: 60,
    height: 60
  },

  optionsNavgation: {
    backgroundColor: '#F2F0F4',
    borderRadius: 10,
    width: '80%',
    paddingLeft: 40,
    paddingBottom: 45,
    marginTop: 54
  },
  buttonsNavgations: {
    backgroundColor: "#413C45",
    width: '80%',
    height: 34,
    borderRadius: 10,
    marginTop: 35,
    justifyContent: "center",
    alignItems: "center"
  },

  textButtonNavigation: {
    color: '#F2F0F4',
    fontSize: 16,
    fontWeight: "700"
  },

  iconNotifications: {
    width: 75,
    height: 75,
    position: "relative",
    left: 276,
    top: 80
  }
});





export default styles;
