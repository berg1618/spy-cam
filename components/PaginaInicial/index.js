import { Image, Text, TouchableOpacity, View } from "react-native";
import FontAwesomeIcon from 'react-native-vector-icons/FontAwesome';
import React from "react";
import styles from "./styles";
import CustomizedBar from "../CustomizedBar/CustomizedBar";

const PaginaInicial = ({ navigation }) => {
  return (
    <View style={styles.container}>
      <CustomizedBar />
      <View style={styles.header}>
        <Text style={styles.title}>Spy Cam</Text>
      </View>
      <TouchableOpacity style={styles.circleButton}>
        <FontAwesomeIcon name='bell' size={40} color='#413C45' style={{ marginTop: 21, marginHorizontal: 24 }} />
      </TouchableOpacity>
      <View style={styles.main}>
        <View>
          <Image style={styles.imgLogo} source={require("../../assets/iconReconhecimento.png")} />
        </View>

        <View style={styles.optionsNavgation}>

          <TouchableOpacity
            style={styles.buttonsNavgations}
            onPress={() => {
              navigation.navigate("Perfil");
            }}
          >
            <Text style={styles.textButtonNavigation}>Perfil</Text>
          </TouchableOpacity>

          <TouchableOpacity
            onPress={() => {
              navigation.navigate("notificacoes")
            }}
            style={styles.buttonsNavgations}>
            <Text style={styles.textButtonNavigation}>Notificações</Text>
          </TouchableOpacity>

          <TouchableOpacity
            style={styles.buttonsNavgations}
            onPress={() => {
              navigation.navigate("CadastrarRosto");
            }}
          >
            <Text style={styles.textButtonNavigation}>Cadastrar Rosto</Text>
          </TouchableOpacity>

        </View>

      </View>
    </View>
  );
};

export default PaginaInicial;
