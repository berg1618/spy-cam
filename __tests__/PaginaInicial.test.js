import React from 'react';
import renderer from 'react-test-renderer';
import PaginaInicial from '../components/PaginaInicial';

test('testando o componente de pagina inicial', () => {
    const tree = renderer.create(<PaginaInicial />).toJSON();
    expect(tree).toMatchSnapshot();
  });