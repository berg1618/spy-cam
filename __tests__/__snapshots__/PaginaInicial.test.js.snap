// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testando o componente de login 1`] = `
<View
  style={
    {
      "backgroundColor": "#413C45",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "borderBottomColor": "#fff",
        "borderBottomWidth": 2,
        "flexDirection": "row",
        "height": 87,
        "justifyContent": "center",
        "marginTop": 15,
      }
    }
  >
    <Text
      style={
        {
          "color": "#F2F0F4",
          "fontSize": 40,
          "fontWeight": "800",
        }
      }
    >
      Spy Cam
    </Text>
    <Image
      source={
        {
          "testUri": "../../../assets/iconMenu.png",
        }
      }
      style={
        {
          "height": 29,
          "left": 60,
          "position": "relative",
          "top": 4,
          "width": 33,
        }
      }
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "marginTop": 54,
      }
    }
  >
    <View>
      <Image
        source={
          {
            "testUri": "../../../assets/iconReconhecimento.png",
          }
        }
        style={
          {
            "height": 60,
            "width": 60,
          }
        }
      />
    </View>
    <View
      style={
        {
          "backgroundColor": "#F2F0F4",
          "borderRadius": 10,
          "marginTop": 54,
          "paddingBottom": 45,
          "paddingLeft": 40,
          "width": "80%",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#413C45",
            "borderRadius": 10,
            "height": 34,
            "justifyContent": "center",
            "marginTop": 35,
            "opacity": 1,
            "width": "80%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#F2F0F4",
              "fontSize": 16,
              "fontWeight": "700",
            }
          }
        >
          Perfil
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#413C45",
            "borderRadius": 10,
            "height": 34,
            "justifyContent": "center",
            "marginTop": 35,
            "opacity": 1,
            "width": "80%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#F2F0F4",
              "fontSize": 16,
              "fontWeight": "700",
            }
          }
        >
          Notificações
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#413C45",
            "borderRadius": 10,
            "height": 34,
            "justifyContent": "center",
            "marginTop": 35,
            "opacity": 1,
            "width": "80%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#F2F0F4",
              "fontSize": 16,
              "fontWeight": "700",
            }
          }
        >
          Cadastrar rosto
        </Text>
      </View>
    </View>
  </View>
  <Image
    source={
      {
        "testUri": "../../../assets/iconNotificacao.png",
      }
    }
    style={
      {
        "height": 75,
        "left": 276,
        "position": "relative",
        "top": 80,
        "width": 75,
      }
    }
  />
</View>
`;

exports[`testando o componente de pagina inicial 1`] = `
<View
  style={
    {
      "backgroundColor": "#413C45",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "borderBottomColor": "#fff",
        "borderBottomWidth": 2,
        "flexDirection": "row",
        "height": 87,
        "justifyContent": "center",
        "marginTop": 15,
      }
    }
  >
    <Text
      style={
        {
          "color": "#F2F0F4",
          "fontSize": 40,
          "fontWeight": "800",
        }
      }
    >
      Spy Cam
    </Text>
    <Image
      source={
        {
          "testUri": "../../../assets/iconMenu.png",
        }
      }
      style={
        {
          "height": 29,
          "left": 60,
          "position": "relative",
          "top": 4,
          "width": 33,
        }
      }
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "marginTop": 54,
      }
    }
  >
    <View>
      <Image
        source={
          {
            "testUri": "../../../assets/iconReconhecimento.png",
          }
        }
        style={
          {
            "height": 60,
            "width": 60,
          }
        }
      />
    </View>
    <View
      style={
        {
          "backgroundColor": "#F2F0F4",
          "borderRadius": 10,
          "marginTop": 54,
          "paddingBottom": 45,
          "paddingLeft": 40,
          "width": "80%",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#413C45",
            "borderRadius": 10,
            "height": 34,
            "justifyContent": "center",
            "marginTop": 35,
            "opacity": 1,
            "width": "80%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#F2F0F4",
              "fontSize": 16,
              "fontWeight": "700",
            }
          }
        >
          Perfil
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#413C45",
            "borderRadius": 10,
            "height": 34,
            "justifyContent": "center",
            "marginTop": 35,
            "opacity": 1,
            "width": "80%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#F2F0F4",
              "fontSize": 16,
              "fontWeight": "700",
            }
          }
        >
          Notificações
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#413C45",
            "borderRadius": 10,
            "height": 34,
            "justifyContent": "center",
            "marginTop": 35,
            "opacity": 1,
            "width": "80%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#F2F0F4",
              "fontSize": 16,
              "fontWeight": "700",
            }
          }
        >
          Cadastrar rosto
        </Text>
      </View>
    </View>
  </View>
  <Image
    source={
      {
        "testUri": "../../../assets/iconNotificacao.png",
      }
    }
    style={
      {
        "height": 75,
        "left": 276,
        "position": "relative",
        "top": 80,
        "width": 75,
      }
    }
  />
</View>
`;
